### PR TITLE
Issue 6099: Table Segment Key Count

### LINKED ISSUE #6099: Table Segment Entry Count API
**Describe the solution you'd like**
We need an API (exposed via the Wire Protocol) that returns the current number of entries in a Table Segment. The number of entries may be approximate if there are ongoing updates to the table, but should be consistent if those updates are all conditional updates.

### PR DESCRIPTION
**Pull request by andreipaduroiu**
**Change log description**  
Created a new API to retrieve information about a Table Segment:
- Length and Start Offset
- Key Length
- Entry Count

**Purpose of the change**  
 Fixes #6099.

**What the code does**  
- Created new Wire Commands `GetTableSegmentInfo` and `TableSegmentInfo` to request and get info about a Table Segment.
- Wired these up from `TableSegmentImpl` (client-side) via the `PravegaRequestProcessor` and into the `TableStore` endpoint of the Segment Store.
- Implemented this inside the Segment Store as an eventually consistent property (value will converge to accurate result as elements are indexed in the background). 
- Removed never-used Wire Commands and support for merging and sealing Table Segments. This functionality has not been implemented in the past 3 years and is not on our roadmap.

**How to verify it**  
Several unit tests have been added and some have been updated.

### CODE REVIEW

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/BadSegmentTypeException.java [hunk 1]
```diff
@@ -19,21 +19,21 @@
  * Exception that is thrown whenever a segment of the wrong type is accessed (i.e., we want a StreamSegment but were given
  * the name of a Table Segment).
  */
-public class BadSegmentTypeException extends StreamSegmentException {
+public class BadSegmentTypeException extends IllegalArgumentException {
```
Comment (RaulGracia) [inline segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/BadSegmentTypeException.java:22]: What are the side effects of this to other components?

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/TableSegmentStatsRecorder.java [hunk 2]
```diff
@@ -122,6 +130,11 @@ public void iterateKeys(String tableSegmentName, int resultCount, Duration elaps
             public void iterateEntries(String tableSegmentName, int resultCount, Duration elapsed) {
             }
 
+            @Override
+            public void getInfo(String tableSegmentName, Duration elapsed) {
+
```
Comment (RaulGracia) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/TableSegmentStatsRecorder.java:135]: Nit: extra space.

**Review by RaulGracia — APPROVED** [review]
LGTM, just a couple of minor comments.

**Comment by RaulGracia** [comment]
Although apparently not related, this test has failed in this build (as well in others):
```
io.pravega.segmentstore.server.containers.StreamSegmentContainerTests > testTransactionOperations FAILED
    java.lang.AssertionError: getContent() did not return a completed future for segmentSegment_73
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.checkReadIndex(StreamSegmentContainerTests.java:2392)
        at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.checkReadIndex(StreamSegmentContainerTests.java:2368)
        at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.checkReadIndex(StreamSegmentContainerTests.java:2363)
        at io.pravega.segmentstore.server.containers.StreamSegmentContainerTests.testTransactionOperations(StreamSegmentContainerTests.java:1333)
````
Looks like this test is becoming increasingly unreliable.

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/BadSegmentTypeException.java (hunk 1 above)
Comment (andreipaduroiu) [reply segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/BadSegmentTypeException.java:22]: This exception has been there for 3 years but never used anywhere. I have made it extend `IllegalArgumentException` and replaced a single place in the code throwing `IllegalArgumentException` with it. So any existing code will still work as it did, but now we have a specific exception to tell us when we try to access the wrong segment type.

**Pull request merged**